server/chat-plugins/trivia.ts

```diff
@@ -386,29 +386,26 @@ class Trivia extends Rooms.RoomGame {
 
 	addTriviaPlayer(user: User) {
 		if (this.playerTable[user.id]) return this.room.tr('You have already signed up for this game.');
-		for (const id in user.prevNames) {
+		for (const id of user.previousIDs) {
 			if (this.playerTable[id]) return this.room.tr('You have already signed up for this game.');
 		}
 		if (this.kickedUsers.has(user.id)) {
 			return this.room.tr('You were kicked from the game and thus cannot join it again.');
 		}
-		for (const id in user.prevNames) {
+		for (const id of user.previousIDs) {
 			if (this.playerTable[id]) return this.room.tr('You have already signed up for this game.');
 			if (this.kickedUsers.has(id)) return this.room.tr('You were kicked from the game and cannot join until the next game.');
 		}
 
 		for (const id in this.playerTable) {
-			const tarUser = Users.get(id);
-			if (tarUser) {
-				if (tarUser.prevNames[user.id]) return this.room.tr('You have already signed up for this game.');
-
-				const tarPrevNames = Object.keys(tarUser.prevNames);
-				const prevNameMatch = tarPrevNames.some(tarId => (tarId in user.prevNames));
-				if (prevNameMatch) return this.room.tr('You have already signed up for this game.');
-
-				const tarIps = Object.keys(tarUser.ips);
-				const ipMatch = tarIps.some(ip => (ip in user.ips));
-				if (ipMatch) return this.room.tr('You have already signed up for this game.');
+			const targetUser = Users.get(id);
+			if (targetUser) {
+				const isSameUser = (
+					targetUser.previousIDs.includes(user.id) ||
+					targetUser.previousIDs.some(tarId => user.previousIDs.includes(tarId)) ||
+					Object.keys(targetUser.ips).some(ip => (ip in user.ips))
```

Comment:
Should `.ips` also be an array?

Reply:
`User#ips` also tracks the number of connections on each IP, which _seems_ useful, but isn't actually used anywhere in public or private code. I think making it an array should be fine?